// react-statement-blocks/dist/if-BmQrJq6Z.js
import { jsx as l, Fragment as a } from "react/jsx-runtime";
import { Children as E, isValidElement as h } from "react";
const d = (e, r) => {
  if (e === "Unknown")
    throw new Error(
      `Invalid branch type: ${e}. Expected one of: <If/>, <ElseIf/>, <Else/>.`
    );
  if (!r && e !== "If")
    throw new Error(`First branch must be <If/>. Found: <${e}/>.`);
  if (r === "If" && e !== "ElseIf" && e !== "Else")
    throw new Error("<If/> branch must be followed by <ElseIf/> or <Else/>.");
  if (r === "ElseIf" && e !== "ElseIf" && e !== "Else")
    throw new Error(
      "<ElseIf/> branch must be followed by <ElseIf/> or <Else/>."
    );
  if (r === "Else")
    throw new Error("<Else/> branch cannot be followed by another branch.");
}, p = ({ children: e }) => {
  let r = null;
  const i = E.toArray(e), t = [];
  for (const n of i) {
    if (!h(n))
      continue;
    const o = n.type.displayName ?? "Unknown";
    d(o, r), r = o, t.push({
      type: o,
      element: n
    });
  }
  for (const n of t) {
    const { condition: o, children: f } = n.element.props, c = n.type;
    if (c === "If" || c === "ElseIf") {
      if (o)
        return /* @__PURE__ */ l(a, { children: typeof f == "function" ? f(o) : n.element });
    } else
      return n.element;
  }
  return null;
}, s = ({ children: e }) => /* @__PURE__ */ l(a, { children: e }), I = (e) => /* @__PURE__ */ l(s, { branchType: "Else", ...e });
I.displayName = "Else";
const b = (e) => /* @__PURE__ */ l(s, { branchType: "ElseIf", ...e });
b.displayName = "ElseIf";
const m = (e) => /* @__PURE__ */ l(s, { branchType: "If", ...e });
m.displayName = "If";
export {
  p as C,
  I as E,
  m as I,
  b as a
};
