// react-statement-blocks/dist/switch-DLDXqMB7.js
import { jsx as u, Fragment as c } from "react/jsx-runtime";
import { Children as l, isValidElement as s, Fragment as h, memo as p } from "react";
function y(e) {
  return /* @__PURE__ */ u(c, { children: e.children });
}
y.displayName = "Case";
function d(e) {
  return /* @__PURE__ */ u(c, { children: e.children });
}
d.displayName = "Default";
function m(e) {
  const r = [];
  return l.forEach(e, (t) => {
    if (s(t) && t.type === h) {
      const n = l.toArray(
        t.props.children
      );
      r.push(...n);
    } else
      r.push(t);
  }), r;
}
function C(e) {
  let r = !1;
  l.forEach(e, (t) => {
    if (!s(t))
      return;
    const n = t.type.displayName;
    if (n !== "Case")
      if (n === "Default") {
        if (r)
          throw new Error("<Switch> can only have one <Default>.");
        r = !0;
      } else
        throw new Error(
          "<Switch> can only contain <Case> or <Default> components."
        );
  });
}
function w(e) {
  return l.toArray(e).sort((t, n) => {
    if (!s(t) || !s(n))
      return 0;
    const i = t.type.displayName, a = n.type.displayName;
    return i === "Case" && a === "Default" ? -1 : i === "Default" && a === "Case" ? 1 : 0;
  });
}
const D = (e, r) => e === r, N = ({
  value: e,
  children: r,
  is: t = D
}) => {
  const n = m(r);
  C(n);
  const i = w(n);
  for (const a of i) {
    if (!s(a))
      continue;
    if (a.type.displayName === "Case") {
      const { value: o } = a.props;
      if ((Array.isArray(o) ? o : [o]).some((f) => t(e, f)))
        return a;
    } else
      return a;
  }
  return null;
}, A = p(N, (e, r) => e.value !== r.value || e.is !== r.is ? !1 : e.children === r.children);
A.displayName = "Switch";
export {
  y as C,
  d as D,
  A as S
};
